indico/modules/events/editing/client/js/editing/timeline/FileManager/util.js

```diff
@@ -48,33 +54,45 @@ async function uploadFile(file, url, onUploadProgress) {
   }
 }
 
+export async function uploadAnExistingFile(url, file) {
+  try {
+    const {data} = await indicoAxios.post(url, snakifyKeys(file));
+    return data;
+  } catch (e) {
+    handleAxiosError(e);
+    return null;
+  }
+}
+
 /**
  * Upload file using axios while triggering all needed actions to update progress.
  *
  * @param {Function} action - the action to be executed after upload is done
  * @param {String} fileTypeId - the id of the File Type
  * @param {Array} acceptedFiles - the "accepted files" array sent by react-dropzone
- * @param {String} uploadURL - the URL to POST the files to
+ * @param {Function} uploadCallback - the function to be called on file upload
  * @param {Function} dispatch - the dispatch function for reducer actions
- * @param {String?} fileId - the ID of the file to modify, if any
+ * @param {String?} replaceFileId - the ID of the file to modify, if any
  * @param {Function} onError - the function to be called on file upload error
+ * @param {Number} fileId - the id of the file, if any, used to cross-reference uploadables
  */
 export function uploadFiles(
   action,
   fileTypeId,
   acceptedFiles,
-  uploadURL,
+  uploadCallback,
```

Comment:
I don't think `callback` is really fitting here... `uploadFunc`?

Reply:
I've seen `Func` being used multiple times before so, seems like a candidate